File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/repository/config/InspectionClassLoaderTest.java
package com.coupang.springframework.data.requery.repository.config;

import com.coupang.springframework.data.requery.repository.config.InspectionClassLoader;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * InspectionClassLoaderTest
 *
 * @author debop@coupang.com
 * @since 18. 6. 14
 */
public class InspectionClassLoaderTest {

    @Test
    public void shouldLoadExternalClass() throws ClassNotFoundException {

        InspectionClassLoader classLoader = new InspectionClassLoader(getClass().getClassLoader());

        Class<?> isolated = classLoader.loadClass("org.h2.Driver");
        Class<?> included = getClass().getClassLoader().loadClass("org.h2.Driver");

        assertThat(isolated.getClassLoader())
            .isSameAs(classLoader)
            .isNotSameAs(getClass().getClassLoader());

        assertThat(isolated).isNotEqualTo(included);
    }
}

File: spring-data-requery/src/test/kotlin/org/springframework/data/requery/repository/config/InspectionClassLoaderTest.kt
package org.springframework.data.requery.repository.config

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class InspectionClassLoaderTest {

    @Test
    fun `should load external class`() {

        val classLoader = InspectionClassLoader(this.javaClass.classLoader)

        val isolated = classLoader.loadClass("org.h2.Driver")
        val included = javaClass.classLoader.loadClass("org.h2.Driver")

        assertThat(isolated.classLoader)
            .isSameAs(classLoader)
            .isNotSameAs(javaClass.classLoader)

        assertThat(isolated).isNotEqualTo(included)
    }
}